### PR TITLE
Added split_multi_generic.

### PR DESCRIPTION
**Pull request by cseed**
Which exposes the generic split_multi interface.
When we nuke Genotype, I will rename split_multi_generic => split_multi and split_multi => split_multi_hts.
Updated the interface slightly.  The rule I'm using is: v, va, g, etc. all denote the old annotations and newV denotes the new, split variant.

### CODE REVIEW

**Comment by cseed** [comment]
I nuked propagateGQ which was a total hack which had been on the chopping block for quite some time.

File: python/hail/dataset.py [hunk 1]
```diff
@@ -4772,12 +4770,101 @@ def skat(self, variant_keys, single_key, weight_expr, y, x, covariates=[], logis
...
+        - ``sa``: the sample annotations
+        - ``g``: the old genotype annotations
+
+        Note, the variant and genotype annotations are updated for all
+        output variants, even those that were originally biallelic.
+        This is because the updates can change the variant annotation
+        and genotype schemas, which must be consistent across
+        variants.
+
+        :param str variant_expr: Annotation expressions to update the variant annotations for the new, split variants.
```
Comment (konradjk) [inline python/hail/dataset.py]: Should this and `genotype_expr` be allowed to take lists?

Comment (tpoterba) [reply python/hail/dataset.py]: I think they can take multiple exprs, comma-delimited. The interface will probably look different in 0.2, so it's not worth making it marginally nicer by taking lists right now!

**Comment by cseed** [comment]
Fixed.  I'll debug the tests shortly.

**Comment by konradjk** [comment]
Will `filter_alleles_generic` go into this PR, or gonna make a separate one?

**Comment by cseed** [comment]
It will be a "stacked" PR on top of this one.  So you'll be able to build that one get get all the generic *_allele functions.

**Pull request merged**